script/tool/test/common/xcode_test.dart

```diff
@@ -94,8 +94,9 @@ void main() {
         }
       };
 
-      processRunner.processToReturn = MockProcess.succeeding();
-      processRunner.resultStdout = jsonEncode(devices);
+      processRunner.mockProcessesForExecutable['xcrun'] = <io.Process>[
```

Comment:
`mockProcessesForExecutable` is something I added fairly recently; previously there was just `processToReturn` which was used for *every* call made during that test. That meant it was just impossible to do things like test the case where the first thing a command ran succeeded and the second failed. Or where the output of two processes was used in the same command. All tests that did process mocking looked like this. Unsurprisingly, we had a ton of test coverage holes as a result. (And it was really hard to follow what process mock results were intended to be for.)

When I added it (to enable writing a bunch of new tests that we should have had in the first place) I didn't go back and clean up all the old cases because I wanted to limit the scope of the PR; this is finally doing that cleanup.